src/main/java/de/tum/in/www1/artemis/web/rest/metis/conversation/ChannelResource.java

```diff
@@ -72,6 +74,9 @@ public ChannelResource(ChannelService channelService, ChannelRepository channelR
     @PreAuthorize("hasRole('USER')")
     public ResponseEntity<List<ChannelDTO>> getCourseChannelsOverview(@PathVariable Long courseId) {
         log.debug("REST request to all channels of course: {}", courseId);
+        if (!(courseRepository.isMessagingEnabled(courseId))) {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN, "Messaging is not enabled for this course");
+        }
```

Comment:
This is duplicated many times. Can you move it into a small helper method to make future changes to the check easier to implement?
```suggestion
        checkMessagingEnabledElseThrow(courseId);
```